components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/ServiceURL.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.core;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Service URL interface.
+ */
+public interface ServiceURL {
+
+    /**
+     * This method is called to get the protocol of Service URL.
+     *
+     * @return String of the protocol.
+     */
+    String getProtocol();
+
+    /**
+     * This method is called to get the host name of Service URL.
+     *
+     * @return String of the host name.
+     */
+    String getHostName();
+
+    /**
+     * This method is called to get the port of Service URL.
+     *
+     * @return value of the port.
+     */
+    int getPort();
+
+    /**
+     * This method is called to get the Url path of Service URL.
+     *
+     * @return String of the url path.
+     */
+    String getUrlPath();
```

Comment:
Would it make sense to name this as getPath()?
getPath() is a commonly used https://docs.oracle.com/javase/7/docs/api/java/net/URL.html#getPath()

Reply:
+1 to use getPath()